### PR TITLE
Newsletter #2: Final draft

### PR DESCRIPTION
**Pull request by ozkriff**
[**Rendered**](https://github.com/ozkriff/rust-gamedev.github.io/blob/newsletter_002_draft/_drafts/newsletter-002.md)

Sorry, I started working on the draft late this time. I was planning to prepare an extended draft during 20ths of September but unexpectedly got totally swamped.

Ideally, this newsletter should be released on 1 October but I hope it's ok if it will be released a few days later this time.

------

TODO:

- [x] Roguelike Toolkit 3.1 & Rust Roguelike Tutorial
- [x] Veloren
- [x] Godot
- [x] Zemeroth
- [x] Use prebuilt rooms with Rust macros for more interesting procedural dungeons
- [x] Embark / texture-synthesis
  - [x] texture-synthesis
  - [x] other
- [x] Iced (pum-purum-pum-pum)
- [x] gfx-rs
- [x] Twenty Asteroids
- [x] Amethyst
  - [x] tooling
  - [x] games
- [x] Hot Reloading
- [x] Other News
  - [x] uset
  - [x] WraithDefense
  - [x] cubism-rs
  - [x] Stats And Skills
  - [x] Robo Instructus
  - [x] Github Actions CI with Rust and SDL2
  - [x] Tennis Academy
  - [x] Garden
  - [x] seratonik's project
  - [x] doom BSP talk
  - [x] rx (cloudhead)
  - [x] Xprite
  - [x] live-rust
  - [x] luminance
  - [x] spline-editor
  - [x] MrVallentin's voxel engine
  - [x] Disconnected
  - [x] minimum
  - [x] imgui-inspect
  - [x] deios
  - [x] blend
- [x] Popular Workgroup Issues in Github
- [x] Requests for Contribution
- [x] Bonus: Sandspiel

### CODE REVIEW

**Comment by 17cupsofcoffee** [comment]
Looking good so far 😄

Take your time - I think maybe once this issue is out we should raise an issue to discuss how we can get more people submitting their own stuff/news that interests them to the newsletter. Feel bad that all of the work is falling on you at the minute!

**Comment by cloudhead** [comment]
I'd love to do the section on `rx`, but probably best to wait for this to be merged first.

**Comment by ozkriff** [comment]
@cloudhead I'm not sure how long finishing and merging this will take, so I will be happy if you make a PR to [this branch](https://github.com/ozkriff/rust-gamedev.github.io/tree/newsletter_002_draft). :)

**Comment by hecrj** [comment]
I would love to help with the Iced section too! I can open a PR with a short description.

**Comment by phaazon** [comment]
That’s a hella lot of work. Congrats! 🎉

**Comment by cloudhead** [comment]
@ozkriff https://github.com/ozkriff/rust-gamedev.github.io/pull/2 :+1:

**Comment by ozkriff** [comment]
The PR is finally ready for review!

r? everyone

If everything is ok, this newsletter will be merged and published in a few hours (~ 16:00 UTC?).

File: _drafts/newsletter-002.md [hunk 1]
```diff
@@ -42,6 +40,372 @@ _Discussions:
...
 -->
 
+### Rust [Roguelike Toolkit][rl-toolkit] and [Roguelike Tutorial][rl-book]
+
+![Minimal pathfinding and FoV example](/assets/newsletter-002/rltk-rs-example-06.gif)
+
+[rltk_rs] by [@herberticus] is a Rust implementation of
+[C++ Roguelike Toolkit][RLTK] ([what is a "roguelike?"][rl]).
+
+It provides all the basic functionality one needs to write a roguelike game,
```
Comment (Wodann) [inline _drafts/newsletter-002.md]:
```suggestion
It provides all the basic functionality one needs to write a roguelike game:
```

File: _drafts/newsletter-002.md [hunk 2]
```diff
@@ -42,6 +40,372 @@ _Discussions:
...
+mouse support, an embedded resource system and Web Assembly support, and more.
```
Comment (Wodann) [inline _drafts/newsletter-002.md]:
```suggestion
mouse support, an embedded resource system, Web Assembly support, and more.
```

File: _drafts/newsletter-002.md [hunk 3]
```diff
@@ -42,6 +40,372 @@ _Discussions:
...
+
+[Zemeroth]: https://github.com/ozkriff/zemeroth
+[zemeroth-v0-6-text]: http://ozkriff.games/2019-09-21--devlog-zemeroth-v0-6
+[Zemeroth v0.6]: https://github.com/ozkriff/zemeroth/releases/tag/v0.6.0
+
+### [Use Prebuilt Rooms with Rust Macros for More Interesting Procedural Dungeons][proc_rooms]
+
+![Top-down view on a generated dungeon](/assets/newsletter-002/proc-dungeons.png)
+
+[@whostolemyhat](https://twitter.com/whostolemyhat) published fourth part
```
Comment (Wodann) [inline _drafts/newsletter-002.md]:
```suggestion
[@whostolemyhat](https://twitter.com/whostolemyhat) published the fourth part
```

File: _drafts/newsletter-002.md [hunk 4]
```diff
@@ -42,6 +40,372 @@ _Discussions:
...
+of their series of tutorials on procedural generation with Rust.
```
Comment (Wodann) [inline _drafts/newsletter-002.md]:
```suggestion
of their tutorial series on procedural generation with Rust.
```

File: _drafts/newsletter-002.md [hunk 5]
```diff
@@ -42,6 +40,372 @@ _Discussions:
...
+In this tutorial, the room generation is updated so it can also pick from a selection
```
Comment (Wodann) [inline _drafts/newsletter-002.md]:
```suggestion
In this tutorial, the room generation is updated so it can pick from a selection
```

**Comment by nico-abram** [comment]
Found a typo at the end
`The source code is aviable on GitHub`

File: _drafts/newsletter-002.md (hunk 1 above)
Comment (ozkriff) [reply _drafts/newsletter-002.md]: This is supposed to be just the first item of the list. Maybe this should be rephrased somehow?  Hmm

**Comment by ozkriff** [comment]
@kvark I intentionally intermixed games and libraries but I guess I'm ok with adding more structure to the newsletter.

I've got a few questions though:
1) What comes first - libraries or games?
2) Should the Amethyst section be split into `Library & Tooling updates/Amethyst` and `Game updates/Amethyst Games`? Amethyst's logo, I guess, stays in the `Library & Tooling` part.
3) Should the "Other News" section be split into `Library & Tooling updates/Other News` and `Game updates/Other News`?

**Comment by Lokathor** [comment]
Games come first. They're what we're ultimately trying to promote. Libraries only exist in service to games.

**Comment by ozkriff** [comment]
@kvark @Lokathor  @17cupsofcoffee restructured and resorted as proposed. :thinking: 

r?

**Comment by ozkriff** [comment]
I guess I should wait for a few more review approvals before merging after all this section moves.

**Comment by AlexEne** [comment]
Thanks for the ton of work you did on this one

File: _drafts/newsletter-002.md (hunk 1 above)
Comment (Wodann) [reply _drafts/newsletter-002.md]: If that's the case, you can just leave the comma. Alternatively, you can try:

**Comment by 17cupsofcoffee** [comment]
New structure looks good to me :)

**Pull request merged**